### DISCUSSION TITLE
在这插件机制基础上怎么引入其他第三方插件包

### DISCUSSION THREAD

**Discussion by pinsengfahaoyinzheng** [yansongda/pay#843] [Q&A]
这里面包含了支付宝和微信的，想要在引入其他支付类型非原生，在这个包的插件机制基础上可以再扩展其他的第三方支付包吗？

**Comment by yansongda** [comment]
可以。

按照源代码里照葫芦画瓢就行

**Comment by pinsengfahaoyinzheng** [comment]
感谢您百忙之中抽空回答我的问题，非常感谢，我想问下插件是不是都在Plugin文件夹下面管理，如果我想引入其他包，是不是需要放到Plugin文件夹下然后所有文件名称改成plugin为后缀，以命名空间查找就行，除了Plugin文件夹外还需要做其他处理吗？再次感谢您的慷慨解答。🤞

**Comment by pinsengfahaoyinzheng** [comment]
您好，请问下想用现有的支付宝和微信支付，如果还想另外再加上其他的第三方支付包，以插件的方式可以平滑的安装到本项目中一起组合使用吗？非常感谢！

**Comment by yansongda** [comment]
可以的。

您自己可以先实际试下

**Comment by pinsengfahaoyinzheng** [comment]
您好，如果采用本项目的插件机制是不是必须要修改文件进行调试？想请问下您有没有简便的方式，比如命令安装或者打包放入？感谢您的解答。

**Comment by yansongda** [comment]
可以自己写插件，然后用你自己写的插件。

详细看下这个：https://pay.yansongda.cn/docs/v3/kernel/plugin.html

**Comment by pinsengfahaoyinzheng** [comment]
那明白了，再次感谢您的解答。万分感谢!
